File: Aditya-Pratap/Sorting/1465.py
class Solution:
    def maxArea(self, h: int, w: int, horizontalCuts: List[int], verticalCuts: List[int]) -> int:

        horizontalCuts = sorted(horizontalCuts)
        verticalCuts = sorted(verticalCuts)

        max_width = horizontalCuts[0]
        max_len = verticalCuts[0]

        for i in range(len(horizontalCuts)-1):
            if horizontalCuts[i+1] - horizontalCuts[i] > max_width:
                max_width = horizontalCuts[i+1] - horizontalCuts[i]
        
        if h - horizontalCuts[-1] > max_width:
                max_width = h - horizontalCuts[-1]
        
        for i in range(len(verticalCuts)-1):
            if verticalCuts[i+1] - verticalCuts[i] > max_len:
                max_len = verticalCuts[i+1] - verticalCuts[i]

        if w - verticalCuts[-1] > max_len:
                max_len = w - verticalCuts[-1]

        print(max_len)
        print(max_width)
        return (max_len * max_width) % (10**9 + 7)
        

File: [Manya]/Leetcode/Sorting/1465.java
package Sorting;

import java.util.Arrays;

class Solution {
    public int maxArea(int h, int w, int[] horizontalCuts, int[] verticalCuts) {
        Arrays.sort(horizontalCuts);
        Arrays.sort(verticalCuts);
        long maxH = Math.max(horizontalCuts[0], h-horizontalCuts[horizontalCuts.length - 1]);
        long maxW = Math.max(verticalCuts[0], w-verticalCuts[verticalCuts.length - 1]);
        
        for(int i = 1; i < horizontalCuts.length; i++){
            maxH = Math.max(horizontalCuts[i]-horizontalCuts[i-1], maxH);
        }
        for(int i = 1; i < verticalCuts.length; i++){
            maxW = Math.max(verticalCuts[i]-verticalCuts[i-1], maxW);
        }
        long area = (maxH * maxW)% 1000000007;
        return (int)(area );
    }
}
